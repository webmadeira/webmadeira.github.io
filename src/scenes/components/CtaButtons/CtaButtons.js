import React from 'react'
import styled from 'styled-components'
import Button from '../../../components/Button/Button'

const ButtonsWrapper = styled.div`
  margin-top: 6rem;
  position: absolute;
  width:100%;
  display: flex;
  justify-content: center;
  background-color: transparent;

   > :first-child {
    margin-right: 2.5rem;
  }
`


const emailOrganizer = (message) => {
  window.location = 'mailto:' // eslint-disable-line
  + 'webmadeira.org@gmail.com'
  + `?subject:Web Madeira ${message}`
}

const CtaButtons = () => (
  <ButtonsWrapper>
    <Button message="BECOME A SPONSOR" callback={() => { emailOrganizer('sponsorship') }} />
    <Button message="BECOME A SPEAKER" callback={() => { emailOrganizer('speaker') }} />
  </ButtonsWrapper>
)

export default CtaButtons
