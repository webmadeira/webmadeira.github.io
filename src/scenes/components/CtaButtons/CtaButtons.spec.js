import React from 'react'
import { shallow } from 'enzyme'
import CtaButtons from './CtaButtons'

describe('CtaButtons component', () => {
  it('must render properly', () => {
    const component = shallow(<CtaButtons />)

    expect(component).toMatchSnapshot()
  })
})
