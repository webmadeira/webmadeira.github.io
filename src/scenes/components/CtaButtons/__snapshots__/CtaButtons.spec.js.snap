// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CtaButtons component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CtaButtons />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Button
          callback={[Function]}
          message="BECOME A SPONSOR"
        />,
        <Button
          callback={[Function]}
          message="BECOME A SPEAKER"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "callback": [Function],
          "message": "BECOME A SPONSOR",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "callback": [Function],
          "message": "BECOME A SPEAKER",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Button
            callback={[Function]}
            message="BECOME A SPONSOR"
          />,
          <Button
            callback={[Function]}
            message="BECOME A SPEAKER"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "callback": [Function],
            "message": "BECOME A SPONSOR",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "callback": [Function],
            "message": "BECOME A SPEAKER",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
